### DISCUSSION TITLE
`jax._src.core.check_scalar_conversion`

### DISCUSSION THREAD

**Discussion by lfaucheux** [jax-ml/jax#23648] [General] [answered]
Is there a specific reason why [`jax._src.core.check_scalar_conversion`](https://github.com/google/jax/blob/main/jax/_src/core.py#L661) relies on `~.ndim` instead of, say, `~.size` ?

Why prevent, for instance, float conversion of arrays that can still be converted to scalars using `~.item()` ?

**Comment by jakevdp** [comment, marked as answer]
Thanks for the question! JAX strives as much as possible to semantically match the latest release of NumPy. We added this particular code after NumPy deprecated scalar conversions for arrays that are not zero-dimensional; see https://github.com/numpy/numpy/pull/10615.
